inference/huggingface/test-opt.py

```diff
@@ -0,0 +1,49 @@
+import os
+import torch
+import deepspeed
+import deepspeed.module_inject as module_inject
+
+from transformers import pipeline, AutoTokenizer, AutoModelForCausalLM
+from transformers.models.opt.modeling_opt import OPTDecoderLayer
+
+# Get local gpu rank from torch.distributed/deepspeed launcher
+local_rank = int(os.getenv('LOCAL_RANK', '0'))
+world_size = int(os.getenv('WORLD_SIZE', '1'))
+
+print(
+    "***************** Creating model in RANK ({0}) with WORLD_SIZE = {1} *****************"
+    .format(local_rank, world_size))
+
+tokenizer = AutoTokenizer.from_pretrained("facebook/opt-125m")
+model = AutoModelForCausalLM.from_pretrained("facebook/opt-125m")
+
+# tokenizer = AutoTokenizer.from_pretrained("facebook/opt-350m")
+# model = AutoModelForCausalLM.from_pretrained("facebook/opt-350m")
+
+# tokenizer = AutoTokenizer.from_pretrained("facebook/opt-1.3b")
+# model = AutoModelForCausalLM.from_pretrained("facebook/opt-1.3b")
+
+# tokenizer = AutoTokenizer.from_pretrained("facebook/opt-2.7b")
+# model = AutoModelForCausalLM.from_pretrained("facebook/opt-2.7b")
+
+generator = pipeline('text-generation',
+                     model=model,
+                     tokenizer=tokenizer,
+                     device=local_rank)
+
+# NOTE: For OPT models the model.config is passed to the policy, so
+# the policy can determine whehter the layer norm is applied
+# before (opt-125m, opt-1.3b, ...) or after (opt-350m) the attention layer.
+ds_opt_policy = module_inject.replace_policy.HFOPTLayerPolicy
+ds_opt_policy.hf_model_config = model.config
+
+injection_policy = {OPTDecoderLayer: ds_opt_policy}
+
+generator.model = deepspeed.init_inference(generator.model,
+                                           mp_size=world_size,
+                                           dtype=torch.half,
+                                           injection_policy=injection_policy,
```

Comment:
Can we change this to use the `'auto'` approach? then the user doesn't have to pass into the policy/config. Otherwise looks good though!

Reply:
thanks @jeffra for reminding this.  updated the PR to auto approach.